stock_available_sale/models/product_product.py

```diff
@@ -0,0 +1,113 @@
+        for product in self:
+            product.immediately_usable_qty -= product.quoted_qty
+
+    def _compute_quoted_qty(self):
+        """Compute the quantities in Quotations."""
+        domain = [('state', '=', 'draft'),
+                  ('product_id', 'in', self.ids)]
+        #  Limit to a specific company
+        if self.env.context.get('force_company', False):
+            domain.append(
+                ('company_id', '=', self.env.context['force_company']))
+        # when we search locations, should children be searched too?
+        if self.env.context.get('compute_child', True):
+            loc_op = 'child_of'
+        else:
+            loc_op = 'in'
+        # Limit to some locations
+        # Take warehouses that have these locations as stock locations
+        if self.env.context.get('location', False):
+            # Search by ID
+            if isinstance(self.env.context['location'], (int, long)):
+                domain.append(
+                    ('order_id.warehouse_id.lot_stock_id', loc_op,
+                     [self.env.context['location']]))
+            # Search by name
+            elif isinstance(self.env.context['location'], basestring):
+                location_ids = [
+                    l.id
+                    for l in self.env['stock.location'].search([
+                        ('complete_name', 'ilike',
+                         self.env.context['location'])])]
+                domain.append(
+                    ('order_id.warehouse_id.lot_stock_id', loc_op,
+                     location_ids))
+            # Search by whatever the context has - probably a list of IDs
+            else:
+                domain.append(
+                    ('order_id.warehouse_id.lot_stock_id', loc_op,
+                     self.env.context['location']))
+        # Limit to a warehouse
+        if self.env.context.get('warehouse', False):
+            domain.append(
+                ('order_id.warehouse_id', '=', self.env.context['warehouse']))
+        # Limit to a period
+        from_date = self.env.context.get('from_date', False)
+        to_date = self.env.context.get('to_date', False)
+        if from_date:
+            domain.extend([
+                ('order_id.requested_date', '>=', from_date),
+                '&',  # only consider 'date' when 'requested_date' is empty
+                ('order_id.requested_date', '=', False),
+                ('order_id.date', '>=', from_date),
+                ])
+        if to_date:
+            domain.extend([
+                ('order_id.requested_date', '<=', to_date),
+                '&',  # only consider 'date' when 'requested_date' is empty
+                ('order_id.requested_date', '=', False),
+                ('order_id.date', '<=', to_date),
+                ])
+        # Compute the quoted quantity for each product
+        results = Counter()
+        for group in self.env['sale.order.line'].read_group(
```

Comment:
@chienandalu As there is a performance problem in Odoo which maybe will not be solved https://github.com/odoo/odoo/issues/18679

It is recommended to execute read_group for computation purpose with `with_context(lang='')`